src/averageAges.js

```diff
@@ -55,7 +79,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => person.sex === 'm' && people.some(woman =>
+      woman.name === person.mother))
+    : people
+      .filter(person => people.some(woman => woman.name === person.mother));
```

Comment:
About comments above, something like that as here.
But as for me better to check onlyWithSon insilde callback function.
Will be less repeating code.